src/app/mods/health-bar/health-bar.ts

```diff
@@ -1,153 +1,148 @@
-                margin-top: 10px;
-            }
+    startMod(): void {}
 
-            .lifeBar {
-                transition-duration: 300ms;
-                height: 100%;
-                width: 0%;
-                background-color: #333;
-            }
+    private bars: { [fighterId: number]: Bar; } = { };
+    private container: HTMLDivElement;
 
-            .shieldBar {
-                transition-duration: 300ms;
-                height: 100%;
-                width: 0%;
-                margin-left: 50%;
-                background-color: #944ae0;
-                position: absolute;
-                top: 0;
-            }
+    constructor(
+        wGame: any,
+        settings: SettingsService,
+        translate: TranslateService
+    ){
+        super(wGame,settings,translate);
 
-            .lifePointsText {
-                font-size: 12px;
-                position: absolute;
-                width: 80px;
-                color: white;
-                text-shadow: 0 0 5px rgba(0, 0, 0, 0.9);
-                transition-duration: 500ms;
-                margin-top: 16px;
-                margin-left: 2px;
-            }`;
-            this.wGame.document.getElementsByTagName('head')[0].appendChild(healthbarCss);
+        if (this.settings.option.vip.general.health_bar) this.enableHealthBars();
+    }
 
-            this.shortcutsHelper = new ShortcutsHelper(this.wGame);
-            this.barContainer = new BarContainer(this.wGame);
+    private enableHealthBars(){
+        Logger.info('- enable Health-Bar');
 
+        let healthbarCss = document.createElement('style');
+        healthbarCss.id = 'healthbarCss';
+        healthbarCss.innerHTML = `
 
-            this.removeOnDeath();
-            this.setFightStart();
-            this.displayOnStart();
-            this.stopOnFightEnd();
-            this.stopOnFightStop();
+        .lifeBarsContainer {
+            position: absolute;
+            top: 0;
+            left: 0;
+            pointer-events: none;
+            z-index: 1;
+        }
 
+        .lifeBarContainer {
+            box-sizing: border-box;
+            background-color: #222;
+            height: 6px;
+            width: 80px;
+            position: absolute;
+            border-radius: 3px;
+            overflow: hidden;
+            transition-duration: 500ms;
+            margin-top: 10px;
+        }
 
-            this.shortcutsHelper.bind(this.params.health_bar_shortcut, () => {
-                this.barContainer.toggle();
-            });
+        .lifeBar {
+            transition-duration: 300ms;
+            height: 100%;
+            width: 100%;
+            background-color: #333;
         }
-    }
 
+        .lifePointsText {
+            font-size: 12px;
+            position: absolute;
+            width: 80px;
+            color: white;
+            text-shadow: 0 0 5px rgba(0, 0, 0, 0.9);
+            transition-duration: 500ms;
+            margin-top: 16px;
```

Comment:
Use "top" and "left", not margin

Reply:
Doesn't make sense... Top behavior usually depends on the position and in this case, using "top" I won't get the result I'm looking for

Reply:
Position is absolute, meaning the effect is the same as the margin. As a good practice, use the position properties to position the elements, and use the boxing properties to adjust spacing between elements.

Reply:
I tried to remove to leave the "top" and "left" ( aka removing the "margin-*") and the result is not the same. that's why i said that....Should i be doing anything else besides removing the "margin-*"?

Reply:
In that specific case, it won't be possible to switch to "top" and "left", as the JS code already uses this property to position the element above the canvas (if I'm not mistaken).
To offset the final position, we'll have to either use margins, translates, or to edit the position computed by the JS part.
I haven't yet run the change on Lindo to see the desired result though, there may be a better solution.

Reply:
Does this mean I can leave it the way it is?

Reply:
Then a good practice would be to group the style altogether, or at least leave comments to explain the use case here !